fractal_server/app/routes/admin/v2.py

```diff
@@ -0,0 +1,418 @@
+            pass
+
+    return job
+
+
+@router_admin_v2.patch(
+    "/job/{job_id}/",
+    response_model=JobReadV2,
+)
+async def update_job(
+    job_update: JobUpdateV2,
+    job_id: int,
+    user: User = Depends(current_active_superuser),
+    db: AsyncSession = Depends(get_async_db),
+) -> Optional[JobReadV2]:
+    """
+    Change the status of an existing job.
+
+    This endpoint is only open to superusers, and it does not apply
+    project-based access-control to jobs.
+    """
+    job = await db.get(JobV2, job_id)
+    if job is None:
+        raise HTTPException(
+            status_code=status.HTTP_404_NOT_FOUND,
+            detail=f"Job {job_id} not found",
+        )
+
+    if job_update.status != JobStatusTypeV1.FAILED:
+        raise HTTPException(
+            status_code=status.HTTP_422_UNPROCESSABLE_ENTITY,
+            detail=f"Cannot set job status to {job_update.status}",
+        )
+
+    setattr(job, "status", job_update.status)
+    setattr(job, "end_timestamp", get_timestamp())
+    await db.commit()
+    await db.refresh(job)
+    await db.close()
+    return job
+
+
+@router_admin_v2.get("/job/{job_id}/stop/", status_code=202)
+async def stop_job(
+    job_id: int,
+    user: User = Depends(current_active_superuser),
+    db: AsyncSession = Depends(get_async_db),
+) -> Response:
+    """
+    Stop execution of a workflow job.
+
+    Only available for slurm backend.
+    """
+
+    _check_backend_is_slurm()
+
+    job = await db.get(JobV2, job_id)
+    if job is None:
+        raise HTTPException(
+            status_code=status.HTTP_404_NOT_FOUND,
+            detail=f"Job {job_id} not found",
+        )
+
+    _write_shutdown_file(job=job)
+
+    return Response(status_code=status.HTTP_202_ACCEPTED)
+
+
+@router_admin_v2.get(
+    "/job/{job_id}/download/",
+    response_class=StreamingResponse,
+)
+async def download_job_logs(
+    job_id: int,
+    user: User = Depends(current_active_superuser),
+    db: AsyncSession = Depends(get_async_db),
+) -> StreamingResponse:
+    """
+    Download job folder
+    """
+    # Get job from DB
+    job = await db.get(JobV2, job_id)
+    if job is None:
+        raise HTTPException(
+            status_code=status.HTTP_404_NOT_FOUND,
+            detail=f"Job {job_id} not found",
+        )
+    # Create and return byte stream for zipped log folder
+    PREFIX_ZIP = Path(job.working_dir).name
+    zip_filename = f"{PREFIX_ZIP}_archive.zip"
+    byte_stream = _zip_folder_to_byte_stream(
+        folder=job.working_dir, zip_filename=zip_filename
+    )
+    return StreamingResponse(
+        iter([byte_stream.getvalue()]),
+        media_type="application/x-zip-compressed",
+        headers={"Content-Disposition": f"attachment;filename={zip_filename}"},
+    )
+
+
+@router_admin_v2.patch(
+    "/task-v1/{task_id}/",
+    status_code=status.HTTP_200_OK,
+)
+async def flag_task_v1_as_v2_compatible(
+    task_id: int,
+    is_v2_compatible: bool = True,
```

Comment:
Let's go without a default here. If the admin calls this endpoint, they must say what's the intended outcome.